zpy/client.py

```diff
@@ -183,25 +187,34 @@ def preview(dataset_config: DatasetConfig, num_samples=10):
 
 @add_newline
 def generate(
-    name: str, dataset_config: DatasetConfig, num_datapoints: int, materialize=False
+    dataset_config: DatasetConfig, num_datapoints: int = 10, materialize: bool = False
 ):
     """
     Generate a dataset.
-
     Args:
-        name: Name of the dataset. Must be unique per Project.
         dataset_config: Specification for a Sim and its configurable parameters.
         num_datapoints: Number of datapoints in the dataset. A datapoint is an instant in time composed of all
                               the output images (rgb, iseg, cseg, etc) along with the annotations.
         materialize: Optionally download the dataset.
     Returns:
         None
     """
+    # https://www.doc.ic.ac.uk/~nuric/coding/how-to-hash-a-dictionary-in-python.html
+    config_json = json.dumps(
+        dataset_config.config,
+        sort_keys=True,
+    )
+    dhash = hashlib.md5()
+    encoded = config_json.encode()
+    dhash.update(encoded)
+    config_hash = dhash.hexdigest()
```

Comment:
can you put this into the DatasetConfig class? i'm thinking 
`@property 
def hash(self):
    # do hash and return`